tests/test_contact_specific_functions.py

```diff
@@ -116,115 +152,58 @@ def test_claculate_contact_forces_rod_rigid_body():
-        ]
+        cylinder.position[..., 0] - 0.5 * cylinder.length * cylinder.director[2, :, 0]
     )
 
     "initializing constants"
+    """Setting contact_k = 1 and other parameters to 0,
+    so the net forces becomes a function of contact forces only."""
     k = 1.0
-    nu = 0.5
-    velocity_damping_coefficient = 0.1
-    friction_coefficient = 0.2
+    nu = 0
+    velocity_damping_coefficient = 0
+    friction_coefficient = 0
 
     "Function call"
     _calculate_contact_forces_rod_rigid_body(
         rod_element_position,
-        rod_length_collection * rod_tangent_collection,
-        cylinder_position[..., 0],
+        rod.length_collection * rod.tangent_collection,
+        cylinder.position[..., 0],
         x_cyl,
-        cylinder_length * cylinder_director[2, :, 0],
-        rod_radius_collection + cylinder_radius,
-        rod_length_collection + cylinder_length,
-        rod_internal_forces,
-        rod_external_forces,
-        cylinder_external_forces,
-        cylinder_external_torques,
-        cylinder_director[:, :, 0],
-        rod_velocity_collection,
-        cylinder_velocity_collection,
+        cylinder.length * cylinder.director[2, :, 0],
+        rod.radius_collection + cylinder.radius,
+        rod.length_collection + cylinder.length,
+        rod.internal_forces,
+        rod.external_forces,
+        cylinder.external_forces,
+        cylinder.external_torques,
+        cylinder.director[:, :, 0],
+        rod.velocity_collection,
+        cylinder.velocity_collection,
         k,
         nu,
         velocity_damping_coefficient,
         friction_coefficient,
     )
 
     "Test values"
+    """The two systems were placed such that they are penetrating and
+    resulting forces act along the x-axis only.
+    The net force was calculated by halving the contact force i.e net force = 0.5 * contact force = -0.25;
+                                                                    where, contact force = k(1) * penetration depth(-1) * gamma(0.5) = -0.5
+    The net force is then divided to the nodes of the rod and the cylinder as per indices."""
+    assert_allclose(cylinder.external_forces, np.array([[-0.5], [0], [0]]))
+    assert_allclose(cylinder.external_torques, np.array([[0], [0], [0]]))
     assert_allclose(
-        cylinder_external_forces,
-        np.array([[-1.389433], [-0.180767], [1.662106]]),
-        rtol=tol,
-        atol=tol,
-    )
-    assert_allclose(
-        cylinder_external_torques,
-        np.array([[0.159707], [-2.261999], [0.310859]]),
-        rtol=tol,
-        atol=tol,
-    )
-    assert_allclose(
-        rod_external_forces,
-        np.array(
-            [
-                [-1.383582, 1.647523, 2.341712],
-                [-0.350649, -0.154162, -0.257856],
-                [-0.702578, -0.836991, 0.078497],
-            ]
-        ),
-        rtol=tol,
-        atol=tol,
+        rod.external_forces, np.array([[0.16, 0.33], [0, 0], [0, 0]]), atol=1e-2
```

Comment:
Can you compute more digits and increase tolerance slightly, at least 1e-6 range?